chapters/parallel/index.md

```diff
@@ -0,0 +1,76 @@
+
+### Implementation
+
+We will approach this from a top-down perspective. 
+
+#### Inter-parallelism
+
+`Pods` are the highest level of abstraction where we can have parallelism. This is achieved by using the `needs` parameter.
+
+```python
+f = (Flow().add(name='p1', needs='gateway')
+           .add(name='p2', needs='gateway')
+           .add(name='p3', needs='gateway')
+           .needs(['p1','p2', 'p3'], name='r1').plot())
+```
+
+The plot of the above topology looks like this:
+
+![](./simple-plot3.svg)
+
+In the above, `p1`, `p2`, `p3` now subscribe to `gateway` and conduct their work in parallel. The last `.needs()` blocks all Pods until they finish their work.
+
+#### Intra-parallelism
+
+Peas are the next level at which we can configure parallelism. In this case, `Pod` receives the data (in its `HeadPea`) and then distributes it to its `Peas`. You can achieve this by using the `parallel` parameter:
+
+
+```python
+f = (Flow().add(name='p1', needs='gateway')
+           .add(name='p2', needs='gateway')
+           .add(name='p3', parallel=3)
+           .needs(['p1','p3'], name='r1').plot())
+```
+
+![](./simple-plot4.svg)
+
+#### State-less vs state-ful
+
+Depending on the type of `Peas`, `parallel` means different things. In *state-less* `Peas` (`Crafters`, `Encoders` etc.), you *parallelize*, not *shard*. This means the data is not persisted, but it's only processed in a parallel manner. 
+
+On the other hand, *state-ful* `Peas` (Indexers) will store the data across *shards*. This terminology is borrowed from other database systems
+
+For more on sharding in other DB systems (with a focus on traditional SQL), see [here](https://subscription.packtpub.com/book/big_data_and_business_intelligence/9781849516723/1/ch01lvl1sec10/using-sharding-and-data-distribution)
+
+For both of these, you need to set `parallel` to some number higher than `1`, and `polling` to `any`.
+
+##### Index vs Query time Polling
+
+We set `polling` to `any` in order to tell the `HeadPea` to spread the data across its inner `Peas`. This allows us to parallelize. You do this for the Flow you will use to index data. 
+
+However, when querying, you need to make sure to use a Flow where the `Pods` are using `polling: all`. This ensures that the requests are seen by all the `Peas`, and that the respective `id` is found. If we use `polling: any` we risk that the key is not seen by the shard that has the data.
+
+#### Scheduling
+
+Jina offers two types of scheduling:
+
+1. `round_robin` will assign tasks in circular order, without any prioritization. We assume all `Peas` have uniform processing power.
+
+2. `load_balance` will assign tasks to the idle `Peas`.
+
```

Comment:
Shall we add a code snippet on scheduling? Like
```python
from jina.enums import  SchedulerType
f = Flow().add(
        name='p1',
        parallel=10, scheduling=SchedulerType.ROUND_ROBIN)
```

Reply:
that makes sense

Reply:
Good idea. Done